pyiceberg/utils/deprecated.py

```diff
@@ -43,3 +43,18 @@ def new_func(*args: Any, **kwargs: Any) -> Any:
         return new_func
 
     return decorator
+
+
+def deprecation_message(deprecated_in: str, removed_in: str, help_message: Optional[str]) -> None:
+    """Mark properties as deprecated.
```

Comment:
Currently we only use this to deprecate properties. But in the future, we may use it to deprecate other things like parameters. How about just making a general comment here saying that adding this will thrown a deprecation warning?